src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -238,6 +238,14 @@ public override void VisitSchema(ISchema schema)
             var n3 = schema.Subscription?.Name;
             if (n1 == n2 && n1 != null || n1 == n3 && n1 != null || n2 == n3 && n2 != null)
                 throw new InvalidOperationException("The query, mutation, and subscription root types must all be different types if provided.");
+            if (schema.Subscription != null)
+            {
+                foreach (var field in schema.Subscription.Fields.List)
+                {
+                    if (field.StreamResolver == null)
+                        throw new InvalidOperationException($"The field '{field.Name}' of the subscription root type '{schema.Subscription.Name}' must have StreamResolver set. You should set StreamResolver only all root fields of subscriptions.");
```

Comment:
Typo? Should it be `on` instead of `all`?

Reply:
In general, fixing these errors one by one is pretty hard because it requires restarting the application multiple times. I think gathering all the errors and throwing single/aggregated exception after the full validation can provide a better experience. This is usually not a problem during development but is annoying during the version upgrade.

Reply:
Ok let's make that change for v8.  Do you want me to continue with this PR or just add this validation into v8?

Reply:
```suggestion
                        throw new InvalidOperationException($"The field '{field.Name}' of the subscription root type '{schema.Subscription.Name}' must have StreamResolver set.");
```

Reply:
> Ok let's make that change for v8. Do you want me to continue with this PR or just add this validation into v8?

Well, it's not like the schema will work properly if this were to fail validation, so might as well leave it in here.

Reply:
Throwing aggregated error is a job for separate PR

Reply:
#3875